### ISSUE TITLE
Can I fix multi-parts at a time?

### ISSUE THREAD

**Issue by Jiacheng188** [cpflat/log2seq#1]
Hello, and thanks for your log2seq, a meaningful toolkit for log analysis.I wonder if i can fix multi-parts and then split the log using seperators,for instance,
for log:
REQ: curl -g -i -X GET http://controller:9696/v2.0/floatingips.json?fixed_ip_address=10.0.0.4&port_id=5c0e6ba6-6964-4417-ab47-b002c187f081 -H "User-Agent: python-neutronclient" -H "Accept: application/json" -H "X-Auth-Token: {SHA1}8e6966d0b6ba20646b9fe7329e7aeeb022549330" _http_log_request /usr/lib/python2.7/site-packages/keystoneauth1/session.py:248
i want to fix the content surrounded by ",and fix the content like "2.7" and the seperate the log using space and dot.
i tried but failed.Could you please help me to find a way to figure out the problem?
Thank you

**Comment by cpflat** [comment]
Thank you for your report, @Jiacheng188. This tool is designed for people like you!

Unfortunately, the previous version of log2seq has a bug for messages including multiple parenthesis pairs.
This bug is fixed after v0.2.3.

Now I succeeded to parse your sample log by the following code:
```
from log2seq.statement import *

text = 'REQ: curl -g -i -X GET http://controller:9696/v2.0/floatingips.json?fixed_ip_address=10.0.0.4&port_id=5c0e6ba6-6964-4417-ab47-b002c187f081 -H "User-Agent: python-neutronclient" -H "Accept: application/json" -H "X-Auth-Token: {SHA1}8e6966d0b6ba20646b9fe7329e7aeeb022549330" _http_log_request /usr/lib/python2.7/site-packages/keystoneauth1/session.py:248'

pattern_url = r"http(s)?://[a-z0-9.:_=&?/-]+"
pattern_abs_path = r"/([a-z0-9.]+/)+[a-z0-9.]+"

statement_rules = [
    FixParenthesis(['"', '"']),
    FixParenthesis(['"', '"']),
    FixParenthesis(['"', '"']),
    Split(" "),
    Fix(pattern_url),
    Split(":"),
    Fix(pattern_abs_path),
    Split("."),
]

statement_parser = StatementParser(statement_rules)
print(statement_parser.process_line(text)[0])
```

This code will output:
`['REQ', 'curl', '-g', '-i', '-X', 'GET', 'http://controller:9696/v2.0/floatingips.json?fixed_ip_address=10.0.0.4&port_id=5c0e6ba6-6964-4417-ab47-b002c187f081', '-H', 'User-Agent: python-neutronclient', '-H', 'Accept: application/json', '-H', 'X-Auth-Token: {SHA1}8e6966d0b6ba20646b9fe7329e7aeeb022549330', '_http_log_request', '/usr/lib/python2.7/site-packages/keystoneauth1/session.py', '248']`

We need to add multiple FixParenthesis in the rules in this case.
Currently, FixParenthesis does not support recursive search (may be supported in the future).

**Comment by Jiacheng188** [comment]
Thank you for your timely reply！Looking forward to the future version 😁
